lib/syskit/data_service.rb

```diff
@@ -90,10 +90,13 @@ def find_device_attached_to(service = nil)
                     elsif driver_services.size > 1
                         raise ArgumentError, "#{self} handles more than one device, you must specify one of #{driver_services.map(&:name).sort.join(", ")} explicitely"
                     end
-                    service = driver_services.first
+                    service = driver_services.find do |driver_service|
+                        service == driver_service
+                    end
+                    
                 end
 
-                arguments["#{service.name}_dev"]
```

Comment:
> Yes I made that change to not have each a single argument for each device. Where there is on argument per device type, that leads to having unassigned arguments on the task. This in turn prevents the task from being started.

This can be fixed by setting 'nil' as default argument with

~~~
default: nil
~~~

Since syskit verifies that at least one argument is set, that would be fine